rotkehlchen/chain/bitcoin/manager.py

```diff
@@ -0,0 +1,413 @@
+import logging
+from collections import defaultdict
+from collections.abc import Sequence
+from typing import TYPE_CHECKING
+
+from rotkehlchen.chain.bitcoin.constants import BLOCKSTREAM_BASE_URL, MEMPOOL_SPACE_BASE_URL
+from rotkehlchen.chain.bitcoin.types import BitcoinTx, BtcScriptType
+from rotkehlchen.chain.bitcoin.utils import query_apis_via_callbacks, query_blockstream_or_mempool_api, \
+    derive_p2pkh_from_p2pk
+from rotkehlchen.constants.assets import A_BTC
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.db.cache import DBCacheDynamic
+from rotkehlchen.db.history_events import DBHistoryEvents
+from rotkehlchen.fval import FVal
+from rotkehlchen.history.events.structures.base import HistoryEvent
+from rotkehlchen.history.events.structures.types import HistoryEventType, HistoryEventSubType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import ensure_type
+from rotkehlchen.types import BTCAddress, Timestamp, Location, SupportedBlockchain
+from rotkehlchen.utils.misc import satoshis_to_btc, ts_sec_to_ms
+from rotkehlchen.utils.network import request_get_dict, request_get
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def _query_blockstream_or_mempool_balances(
```

Comment:
this function and the one below feel like they should be in the BitcoinManager class.